### PR TITLE
[Opt](Hash) Deduce virtual function call is null at in single nullable column

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Remove unless virtual function call in `HashMethodSingleLowNullableColumn`

Before：

Query group by in clickbench 0.82s

After：

Query group by in clickbench 0.76s

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/common/columns_hashing.h [hunk 1]
```diff
@@ -227,11 +227,11 @@ struct HashMethodSingleLowNullableColumn : public SingleColumnMethod {
     HashMethodSingleLowNullableColumn(const ColumnRawPtrs& key_columns_nullable,
                                       const Sizes& key_sizes, const HashMethodContextPtr& context)
             : Base(get_nested_column(key_columns_nullable[0]), key_sizes, context),
-              key_columns(key_columns_nullable) {}
+              key_column(reinterpret_cast<const ColumnNullable*>(key_columns_nullable[0])) {}
```
Comment (BiteTheDDDDt) [inline be/src/vec/common/columns_hashing.h:230]: Maybe it is better to use `assert_cast` here?

Comment (HappenLee) [reply be/src/vec/common/columns_hashing.h:230]: OK

Comment (HappenLee) [reply be/src/vec/common/columns_hashing.h:230]: OK

**Review by BiteTheDDDDt — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**